lucene/core/src/java/org/apache/lucene/search/IndexSearcher.java

```diff
@@ -193,6 +196,13 @@ public IndexSearcher(IndexReader r, Executor executor) {
     this(r.getContext(), executor);
   }
 
+  /** Same as above with additional Circuit Breaker passed in
+   *
+   * @lucene.experimental */
+  public IndexSearcher(IndexReader r, Executor executor, SliceAllocationCircuitBreaker sliceAllocationCircuitBreaker) {
```

Comment:
I think it'd be cleaner if we combined the `Executor` and this new `SliceAllocationCircuitBreaker` into a single new interface that could both provide the `Executor` and the red-line logic?